### PR TITLE
Create an RSS Feed Edge App

### PR DESCRIPTION
**Pull request by nicomiguelino**
#### Overview

- Resolves Screenly/Playground#45

#### References

- Screenly/Playground#53

#### Checklist

- [x] Write the bare minimum for the RSS Reader Edge App.
- [x] Add a setting for the RSS feed `url`.
- [x] Add a setting for the RSS feed `title`.
- [x] Include a part of the article's body on the page.
- [x] Add a setting for the `limit`.
- [x] Modify `screenly.yml`.
- [x] Use **_Moment.js_** to format date and time.
- [x] Don't include the links. People won't be able to click them.
- [x] Make the `limit` setting a required field.
- Have the `content` field be displayed more intelligently. &mdash; Some have just plain text. Some have `<p></p>` tags, while some have `` tags as well. (This will be done in future iterations.)
- [x] Add support for caching. Make sure to cache items for `$CACHE` minutes, where `$CACHE` is a setting.
- [x] Apply styling according to the mockups provided.
- [x] The Edge App should support the resolutions listed in the [Supported Screen Resolutions](https://github.com/Screenly/Playground/blob/master/docs/resolutions.md) page.
- [ ] Make sure that the Playground theme files will be uploaded as well. (It's currently included in `.gitignore`).

##### With images

image

##### 480 x 800

image

##### 800 x 480

image

##### 720 x 1280

image

##### 1280 x 720

image

##### 1080 x 1920

image

##### 1920 x 1080

image

##### 2160 x 3840

image

##### 3840 x 2160

image

##### 2160 x 4096

image

##### 4096 x 2160

image

### CODE REVIEW

File: edge-apps/rss-reader/index.html [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  <body>
+    <template id="feed-template">
+      <div class="feed-container" style="margin-bottom: 40px;">
+        <h2 class="feed-title"></h2>
+        <p class="feed-date"></p>
+        <a class="feed-link" href="#"></a>
+      </div>
+    </template>
+
+    <h1>RSS Reader</h1>
```
Comment (vpetersson) [inline edge-apps/rss-reader/index.html:16]: Make this a title as a setting.

File: edge-apps/rss-reader/index.html [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <title>Hello world</title>
```
Comment (vpetersson) [inline edge-apps/rss-reader/index.html:5]: Use setting title

File: edge-apps/rss-reader/screenly.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+app_id: 01H5X26SAC194MEJDFSMV7BFTD
+root_asset_id: 01H5X2X3VHA3AJD84TZSHD4ZV9
+user_version: v1.0.0
+revision: 1
+description: Screenly RSS Reader App
+icon: ''
+author: Screenly, Inc.
+homepage_url: ''
+settings:
+  disable_analytics:
```
Comment (vpetersson) [inline edge-apps/rss-reader/screenly.yml:10]: Add setting for RSS feed URL and title for RSS feed.

Comment (nicomiguelino) [reply edge-apps/rss-reader/screenly.yml:10]: Got it, thanks. I've already added a setting for RSS feed URL in this commit. Stay tuned for the other settings.

File: edge-apps/rss-reader/screenly.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  rss_url:
+    type: text
+    default: ''
+    optional: false
+    title: RSS Feed URL
+    help_text: Specify the URL to where to get the data from.
+  limit:
+    type: text
+    default: '1'
+    optional: true
```
Comment (vpetersson) [inline edge-apps/rss-reader/screenly.yml:44]: Not optional as it can't operate without this.

Comment (nicomiguelino) [reply edge-apps/rss-reader/screenly.yml:44]: Acknowledged, thanks.

**Comment by nicomiguelino** [comment]
Hello, @iamplugged. I'd like to request a review for this PR. Here's some front-end issue(s) I've experienced so far:

- Using `xl` and `xxl` breakpoints (in margins, paddings, or `col`s) won't have any effect on larger screens (e.g., 4K).
- The best I could do is write media queries on a CSS file. I;m even considering overriding the breakpoint values.
- For instance, if `limit` is set to `1`, an RSS article doesn't occupy the whole row. (I'm aware that I've added the following classes &mdash; `col-sm-12`, `col-lg-12`, `col-xl-4`). I'm planning to do DOM manipulation in JS (and do `element.classList.add(className)`).

**Comment by L4cko** [comment]
@nicomiguelino 

We need to fix the header line divider. It's to strong currently. Here is the example
https://www.figma.com/file/ib2AYqf3B9lYrJjIa3pivW/Screenly-Playground?type=design&node-id=4-997&mode=design&t=MdZLnwglTwQohJAk-4

For the date text, please add opacity 80%. The heading and paragraph are pure white color.

I'm a little bit worried about the small sizes of the text. It may be hard to read. I will have to create you a guideline for text sizes on all screen sizes.

**Comment by nicomiguelino** [comment]
@L4cko & @vpetersson, I'll wrap up changes to this Edge App for now (as per Viktor's instructions). I'll go ahead and do fixes to the Clock Edge App.

The most recent commit in this PR only supports text. Support for images will be continued in future iterations.

Here's what the recent changes look like on a **_3840 x 2160_** landscape display:

image

File: edge-apps/rss-reader/screenly.js
Comment (nicomiguelino) [inline edge-apps/rss-reader/screenly.js:1]: Note to self: This shouldn't be included in the changes.

**Review by vpetersson — APPROVED** [review]

**Pull request merged**